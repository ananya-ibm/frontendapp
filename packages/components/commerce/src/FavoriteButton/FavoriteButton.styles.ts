/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import styled from 'styled-components';
import { FavoriteFilled } from '@carbon/react/icons';
import theme from './FavoriteButton.theme';

export const FavButton = styled('button')`
  background: none;
  border: none;
  cursor: pointer;
  left: 0;
  margin-left: ${props => theme(props).favoriteSpacing};
  margin-top: ${props => theme(props).favoriteSpacing};
  outline: none;
  position: absolute;
  top: 0;
`;

export const FilledButton = styled(FavoriteFilled).attrs(() => ({ size: 20 }))`
  color: ${props => theme(props).activeColor};
`;
