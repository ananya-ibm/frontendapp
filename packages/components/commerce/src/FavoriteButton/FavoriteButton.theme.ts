/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

type Props = {
  activeColor: string;
  favoriteSpacing: string;
};

declare global {
  interface EXOComponentStyles {
    commerce_favoriteButton?: Partial<Props>;
  }
}

export default (props: EXOThemeProps): Props => {
  return {
    // Default theme
    activeColor: props.theme.colors.interactive.primary.base.bg,
    favoriteSpacing: props.theme.spacing.inline.s4,

    // Override with custom theme
    ...props.theme?.byComponent?.commerce_favoriteButton
  };
};
