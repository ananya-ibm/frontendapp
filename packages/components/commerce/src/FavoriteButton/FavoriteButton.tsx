/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import React, { useState } from 'react';
import { Favorite } from '@carbon/react/icons';
import * as S from './FavoriteButton.styles';

export const FavoriteButton = ({ onChange = () => {}, id = 'test' }: Props) => {
  const [isSelected, setSelected] = useState(false);
  const toggleSelected = () => {
    onChange(id);
    setSelected(!isSelected);
  };

  return (
    <S.FavButton
      data-testid="FavoriteBtnTestId"
      role="button"
      tabIndex={0}
      onKeyPress={toggleSelected}
      onClick={toggleSelected}
    >
      {isSelected ? <S.FilledButton /> : <Favorite size={20} />}
    </S.FavButton>
  );
};

type Props = {
  onChange?: (id: string) => void;
  id?: string;
};
