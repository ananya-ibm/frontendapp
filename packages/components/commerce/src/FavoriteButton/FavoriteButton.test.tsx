/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import React from 'react';
import { renderWithTheme as render } from '@testUtils';
import { screen, fireEvent } from '@testing-library/react';
import { axe } from 'jest-axe';
import { FavoriteButton } from './FavoriteButton';

const mockOnChange = jest.fn();

const testProps = {
  id: 'test',
  onChange: mockOnChange
};

describe('<FavoriteButton /> component', () => {
  test('its snapshot matches', () => {
    const { container } = render(<FavoriteButton {...testProps} />);

    expect(container).toBeDefined();
    expect(container.firstChild).toMatchSnapshot();
  });

  test('clicking the button returns the id of the product', () => {
    render(<FavoriteButton {...testProps} />);
    const favButton = screen.getByTestId('FavoriteBtnTestId');

    fireEvent.click(favButton);
    expect(mockOnChange).toHaveBeenCalledWith('test');
  });

  test('it passes an axe accesibility check', async () => {
    const { container } = render(<FavoriteButton {...testProps} />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
