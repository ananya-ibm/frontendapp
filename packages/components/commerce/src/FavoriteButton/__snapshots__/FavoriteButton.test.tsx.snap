// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FavoriteButton /> component its snapshot matches 1`] = `
.c0 {
  background: none;
  border: none;
  cursor: pointer;
  left: 0;
  margin-left: 0.75rem;
  margin-top: 0.75rem;
  outline: none;
  position: absolute;
  top: 0;
}

<button
  class="c0"
  data-testid="FavoriteBtnTestId"
  role="button"
  tabindex="0"
>
  <svg
    aria-label="icon"
    fill="currentColor"
    focusable="false"
    height="20"
    preserveAspectRatio="xMidYMid meet"
    role="img"
    size="20"
    viewBox="0 0 32 32"
    width="20"
    xmlns="http://www.w3.org/2000/svg"
  />
</button>
`;
